chia/util/keychain.py

```diff
@@ -362,6 +366,36 @@ def add_private_key(self, mnemonic: str, label: Optional[str] = None) -> Private
 
         return key
 
+    def add_public_key(self, pubkey: str, label: Optional[str] = None) -> G1Element:
+        """
+        Adds a public key to the keychain.
+        """
+        key = G1Element.from_bytes(hexstr_to_bytes(pubkey))
+        index = self._get_free_private_key_index()
+        fingerprint = key.get_fingerprint()
+
+        if fingerprint in [pk.get_fingerprint() for pk in self.get_all_public_keys()]:
+            # Prevents duplicate add
+            raise KeychainFingerprintExists(fingerprint)
+
+        # Try to set the label first, it may fail if the label is invalid or already exists.
+        # This can probably just be moved into `FileKeyring.set_passphrase` after the legacy keyring stuff was dropped.
+        if label is not None:
+            self.keyring_wrapper.set_label(fingerprint, label)
+
+        try:
+            self.keyring_wrapper.set_passphrase(
+                self.service,
+                get_private_key_user(self.user, index),
+                bytes(key).hex(),
+            )
```

Comment:
I must admit to being a little confused what set_passphrase is doing here, but I note its used in add_private_key as well where it is equally confusing

Reply:
Ok, this is oldish code to maintain API consistency with the keyring python module. Could be refactored out since we don't put "keys as a passphrase" anymore into the OS keyring...